### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by meggrasse**
Hopefully this is self-documenting enough ;)

Check out `target/docs/wasm_module/index.html` for the doc interface.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,8 +1,10 @@
 #[macro_use]
 mod macros;
 mod either;
+/// WebAssembly module and components.
```
Comment (sarahlim) [inline src/lib.rs]: Should we maybe move these to the top of their respective modules?

File: src/module.rs [hunk 2]
```diff
@@ -14,12 +14,14 @@ use tracer::{EntryKind, EXPOSE_TRACER, EXPOSE_TRACER_LEN, LOG_CALL};
 static VOID_VALUE_PLACEHOLDER: i32 = i32::MAX;
 
 #[derive(Debug)]
+/// WebAssembly module
```
Comment (sarahlim) [inline src/module.rs]: Maybe change this to "wrapper around the parity-wasm `Module` struct, with convenience functions"?

File: src/module.rs [hunk 3]
```diff
@@ -32,23 +34,27 @@ impl WasmModule {
...
 
-    /// Safe function for counting the number of imported functions.
+    /// Counts number of imported functions.
     /// Use this instead of `self.imports().size_hint()`!
     pub fn imported_functions_count(&self) -> usize {
         self.module.import_count(ImportCountType::Function)
     }
 
+    /// Iterates over the imports within the function index space of the module.
+    /// Imports will always have `None` for their `body`.
```
Comment (sarahlim) [inline src/module.rs]: Maybe worth clarifying "Imported *functions* do not have `body`s"

File: src/module.rs [hunk 4]
```diff
@@ -274,10 +279,12 @@ impl WasmModule {
         names
     }
 
+    /// Function name for index of exported function in function index space.
     pub fn get_function_name(&self, id: usize) -> Option<&str> {
         self.function_names.get(&id).map(String::as_str)
     }
 
+    /// Iterates over the type of each function in the module.
```
Comment (sarahlim) [inline src/module.rs]: in the function section (as opposed to the function index space)

**Review by sliminality — APPROVED** [review]

**Comment by sarahlim** [comment]
lgtm!

**Pull request merged**